File: output/deepseek_chat/avatar/algo_based_translation/python/java/failed_translation/atcoder_ABC123_C.java

import java.util.Scanner;
import java.util.Arrays;
import java.util.List;
import java.util.ArrayList;
import java.util.Collections;

public class atcoder_ABC123_C {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int N = scanner.nextInt();
        List<Integer> A = new ArrayList<>();
        for (int i = 0; i < 5; i++) {
            A.add(scanner.nextInt());
        }
        int minVal = Collections.min(A);
        double result = Math.ceil((double) N / minVal);
        System.out.println((int) result + 4);
    }
}

File: output/deepseek_chat/avatar/direct_translation/java/python/successful_translation/atcoder_ABC123_C.py

import math

def main():
    N = int(input())
    min_val = 0
    for i in range(5):
        if i == 0:
            min_val = int(input())
        else:
            min_val = min(min_val, int(input()))
    print(math.ceil(N / min_val) + 4)

if __name__ == "__main__":
    main()

